tests/unit/test_core_lifecycle.py

```
@@ -0,0 +1,532 @@
+def test_bound_loop_mismatch_via_clientcore_raises_runtime_error() -> None:
+    """Cross-loop reuse of a single :class:`ClientCore` raises a clean
+    ``RuntimeError`` on the second loop's first authed POST.
+
+    Reaches through the ``ClientCore`` facade (rather than ``ClientLifecycle``
+    in isolation) because the guard lives in :class:`AuthedTransport` and
+    only fires from inside an authed POST. The test runs two separate
+    ``asyncio.run`` invocations to materialise two distinct loops.
+    """
+    from notebooklm._core import ClientCore
+
+    auth = AuthTokens(csrf_token="CSRF", session_id="SID", cookies={"SID": "v1"})
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to argument: "csrf_token"

(S106)